bl2/bl2_image_load_v2.c

```diff
@@ -87,8 +87,10 @@ entry_point_info_t *bl2_load_images(void)
 	assert(bl2_to_next_bl_params->h.version >= VERSION_2);
 	assert(bl2_to_next_bl_params->head->ep_info);
 
-	/* Populate arg0 for the next BL image */
-	bl2_to_next_bl_params->head->ep_info->args.arg0 = (u_register_t)bl2_to_next_bl_params;
+	/* Populate arg0 for the next BL image if not already provided */
+	if (bl2_to_next_bl_params->head->ep_info->args.arg0 == 0)
```

Comment:
@etienne-lms 
Apologies - still getting to grips with this checker... minor correction here to resolve this:
MISRA C-2012 Rule 10.4 violation: The operands bl2_to_next_bl_params->head->ep_info->args.arg0 and 0 have essential type categories unsigned 64-bit int and signed 8-bit int, which do not match.

Change 0 to 0ULL - sorry, should have spotted that in my previous comment :(